crates/dtrace_probe/src/main.rs

```diff
@@ -28,19 +44,22 @@ fn main() -> Result<()> {
     // We need to set the bufsize option explicitly; otherwise we get "Enabling exceeds size of
     // buffer" error.
     let options = &[("bufsize", "4m")];
-    let pid = args.process_id;
+    let predicate =
+        args.process_id.map(|pid| format!("/pid == {pid}/")).unwrap_or_else(|| "".to_string());
     dtrace.execute_program(
         &format!(
-            "syscall:::entry /pid == {pid}/ {{ trace(walltimestamp); trace(arg0); trace(arg1); }}"
+            "syscall:::entry {predicate} {{ trace(walltimestamp); trace(arg0); trace(arg1); }}"
         ),
         options,
     )?;
 
-    loop {
+    while running.load(Ordering::SeqCst) {
+        let mut message = EventMessage::new();
```

Comment:
I think this can't compile - I guess you meant
```suggestion
        let mut message = EventMessage::default();
```